src/common/catalog/on-context-menu-open.injectable.ts

```diff
@@ -0,0 +1,23 @@
+/**
```

Comment:
The naming could be improved here: `onContextMenuOpen` falsely communicates opportunity for other dynamic code to get delegated an event. 

What we have instead is code that depicts two fixed responsibilities: `openContextMenu` and something weirder like `emitContextMenuOpenIfCategoryExists`. 

Maybe the injectable should be called `openContextMenuVisitor`, thus advertising its visitor-pattern -like nature? There are other ways to improve this design as well, but let's not go there now.

Reply:
Sure I can rename it. I think that this change is already a big improvement over the previous setup where the individual instances would reach out to global registry to get their own category and then do the emitting themselves

Reply:
It is a big improvement :)